_posts/2024-06-13-installing-maven-style-embulk-plugins.md

```diff
@@ -0,0 +1,197 @@
+2024-06-13 15:46:13.027 +0900 [INFO] (main): Downloaded com.amazonaws:aws-java-sdk-pom:pom:1.11.466 at /home/user/.embulk/lib/m2/repository/com/amazonaws/aws-java-sdk-pom/1.11.466/aws-java-sdk-pom-1.11.466.pom
+...
+...
+2024-06-13 15:46:14.857 +0900 [INFO] (main): Downloading org.embulk:embulk-input-s3:jar:0.6.0 from https://repo.maven.apache.org/maven2
+2024-06-13 15:46:14.857 +0900 [INFO] (main): Downloading com.amazonaws:aws-java-sdk-s3:jar:1.11.466 from https://repo.maven.apache.org/maven2
+...
+...
+2024-06-13 15:46:15.720 +0900 [INFO] (main): Downloaded org.embulk:embulk-input-s3:jar:0.6.0 at /home/user/.embulk/lib/m2/repository/org/embulk/embulk-input-s3/0.6.0/embulk-input-s3-0.6.0.jar
+2024-06-13 15:46:15.721 +0900 [INFO] (main): Downloaded com.amazonaws:aws-java-sdk-s3:jar:1.11.466 at /home/user/.embulk/lib/m2/repository/com/amazonaws/aws-java-sdk-s3/1.11.466/aws-java-sdk-s3-1.11.466.jar
+...
+...
+2024-06-13 15:46:15.730 +0900 [INFO] (main): Installed org.embulk:embulk-input-s3:jar:0.6.0 at /home/user/.embulk/lib/m2/repository/org/embulk/embulk-input-s3/0.6.0/embulk-input-s3-0.6.0.jar
+2024-06-13 15:46:15.730 +0900 [INFO] (main): Installed com.amazonaws:aws-java-sdk-s3:jar:1.11.466 at /home/user/.embulk/lib/m2/repository/com/amazonaws/aws-java-sdk-s3/1.11.466/aws-java-sdk-s3-1.11.466.jar
+...
+...
+```
+
+This subcommand downloads also the dependencies of the specified Maven artifact transitively as you can see in the example above.
+
+Note that you can change the destination Embulk home directory by Embulk's standard options. See the example below.
+
+```
+$ java -jar embulk-0.11.3.jar -Xembulk_home=/tmp/foo install "org.embulk:embulk-input-s3:0.6.0"
+...
+
+$ env EMBULK_HOME=/tmp/bar java -jar embulk-0.11.3.jar install "org.embulk:embulk-input-s3:0.6.0"
+...
+```
+
+It now supports only [Maven Central](https://central.sonatype.com/) as the remote repository, unfortunately.
+
+## #2: Out-of-Embulk Embulk plugin installer
+
+Embulk has had the `mkbundle` subcommand and the `-b` option so that users can maintain plugin installations by `Gemfile`, but it works only for RubyGems-style plugins, of course.
+
+[The Gradle `org.embulk.runset` plugin](https://github.com/embulk/gradle-embulk-runset) is an alternative for Maven-style Embulk plugin. It works out of the Embulk package at all.
+
+To use this, set up an environment for [Gradle](https://gradle.org/install/) at first. [Gradle 8.7](https://docs.gradle.org/8.7/userguide/userguide.html) is at least required. You may want to choose [the Gradle wrapper](https://docs.gradle.org/8.7/userguide/userguide.html) in typical use-cases.
+
+Next, write `build.gradle` to declare the Maven-based Embulk plugins you wanted to install.
+
+```
+plugins {
+    id "org.embulk.runset" version "0.2.0"  // Just apply this Gradle plugin.
+}
+
+repositories {
+    mavenCentral()
+}
```

Comment:
[Q] Is this syntax supports non Maven Central Repository?

Reply:
https://github.com/embulk/embulk.github.io/pull/25/files#diff-c097610d9d6b883e779e825fdfae28d317a440ad4c6be732bbadb78410f1f71fR65

Reply:
I believed it's clear as this sentense is located in the "1: Embulk's built-in subcommand `install`" section...

Reply:
>     // It now supports only Maven Central
>     mavenCentral()

At least, the Gradle plugin does not intend to support only Maven Central. It SHOULD work.

Reply:
I see. thank you. It's cleared for me.